examples/intellij-plugin-with-experimental-shared-base/src/main/kotlin/com/jetbrains/compose/panel/CounterPanel.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package com.jetbrains.compose.panel
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.*
+
+@Composable
+fun CounterPanel(stateWithIdeLifecycle: MutableState<CounterState>) {
+    var stateIDE by remember { stateWithIdeLifecycle }
```

Comment:
If we inject state outside of Composable, it is better to not remember it (i.e., just use `stateWithIdeLifecycle`). The user of this Composable could want to inject a new state instance, for example.

Reply:
Done